src/plugins/data_view_management/public/components/index_pattern_table/index_pattern_table.tsx

```diff
@@ -69,6 +71,24 @@ interface Props extends RouteComponentProps {
 
 const getEmptyFunctionComponent: React.FC<SpacesContextProps> = ({ children }) => <>{children}</>;
 
+// shared code, would live elsewhere
+function useStateSelector<S, R>(
```

Comment:
Shared code that simplifies consuming state. Need to verify that this is good and efficient use of hooks.